packages/controls/src/controls/select/select.ts

```diff
@@ -0,0 +1,182 @@
+import { z } from 'zod'
+
+import { map } from '../../lib/functional'
+import { safeParse, unionOfLiterals, type ParseResult } from '../../lib/zod'
+
+import { type CopyContext } from '../../context'
+import {
+  type DeserializedRecord,
+  type SerializedRecord,
+} from '../../serialization'
+
+import { ControlDefinition, serialize, type SchemaType } from '../definition'
+import { DefaultControlInstance, type SendMessage } from '../instance'
+
+type Option<T extends string> = { readonly value: T; readonly label: string }
+type OptionList<T extends string> = readonly [Option<T>, ...Option<T>[]]
```

Comment:
Made `options` a non-empty array: it makes sense to require at least one option, and it allows us to implement Zod's `unionOfLiterals` below w/o hacks.